packages/amp/src/__tests__/amp.test.ts

```diff
@@ -0,0 +1,181 @@
+    expect(normalize("/tag/some/amp/")).toBe("/tag/some/");
+    expect(normalize("/taxonomy/some/amp/")).toBe("/taxonomy/some/");
+
+    expect(normalize("/amp/?a=1&b=1#some-hash")).toBe("/?a=1&b=1#some-hash");
+    expect(normalize("/amp/page/3/?a=1&b=1#some-hash")).toBe(
+      "/page/3/?a=1&b=1#some-hash"
+    );
+    expect(normalize("/taxonomy/some/amp/?a=1&b=1#some-hash")).toBe(
+      "/taxonomy/some/?a=1&b=1#some-hash"
+    );
+    expect(normalize("/taxonomy/some/page/3/?a=1&b=1#some-hash")).toBe(
+      "/taxonomy/some/page/3/?a=1&b=1#some-hash"
+    );
+
+    // These ones are not correct, but it is expected because Frontity does not
+    // support using `amp` as a [term] (https://wordpress.org/support/article/glossary/#term)
+    // for the time being.
+    expect(normalize("/category/amp/")).toBe("/category/");
+    expect(normalize("/tag/amp/")).toBe("/tag/");
+    expect(normalize("/taxonomy/amp/")).toBe("/taxonomy/");
+    expect(normalize("/author/amp/")).toBe("/author/");
+  });
+
+  test(`Compare the output of libraries.source.normalize() between the AMP package and the default implementation`, () => {
+    // Get the reference to normalize() before it's overriden in the init() action.
+    const { normalize } = store.libraries.source;
+    store.actions.amp.init();
+
+    // The implementation of normalize() from the AMP package.
+    const { normalize: normalizeAmp } = store.libraries.source;
+
+    expect(normalize("/")).toEqual(normalizeAmp("/"));
+    expect(normalize("/some-post/")).toEqual(normalizeAmp("/some-post/"));
+    expect(normalize("/some-page/")).toEqual(normalizeAmp("/some-page/"));
+    expect(normalize("/parent/some-page/")).toEqual(
+      normalizeAmp("/parent/some-page/")
+    );
+    expect(normalize("/category/some/")).toEqual(
+      normalizeAmp("/category/some/")
+    );
+    expect(normalize("/tag/some/")).toEqual(normalizeAmp("/tag/some/"));
+    expect(normalize("/taxonomy/some/")).toEqual(
+      normalizeAmp("/taxonomy/some/")
+    );
+    expect(normalize("/?a=1&b=1#some-hash")).toEqual(
+      normalizeAmp("/?a=1&b=1#some-hash")
+    );
+    expect(normalize("/page/3/?a=1&b=1#some-hash")).toEqual(
+      normalizeAmp("/page/3/?a=1&b=1#some-hash")
+    );
+    expect(normalize("/taxonomy/some/?a=1&b=1#some-hash")).toEqual(
+      normalizeAmp("/taxonomy/some/?a=1&b=1#some-hash")
+    );
+    expect(normalize("/taxonomy/some/page/3/?a=1&b=1#some-hash")).toEqual(
+      normalizeAmp("/taxonomy/some/page/3/?a=1&b=1#some-hash")
+    );
+  });
+
+  test("Compare the output of libraries.source.parse() between the AMP package and the default implementation", () => {
+    // Get the reference to parse() before it's overriden in the init() action.
+    const { parse } = store.libraries.source;
+    store.actions.amp.init();
+
+    // The implementation of parse() from the AMP package.
+    const { parse: parseAmp } = store.libraries.source;
+
+    expect(parse("/")).toEqual(parseAmp("/"));
```

Comment:
Hey @michalczaplinski, do you think we should assert that `parse !== parseAmp`? I am thinking that we should make sure that after init with amp package that parse it's not the same anymore. What are your thoughts?

Reply:
Good point, @cristianbote :) I will add that change!

Reply:
ok, done in https://github.com/frontity/frontity/commit/5c66bb90d9b9feb72e4a0384b1185b8b986b1f4b